### PR TITLE
set default num_cpu_threads_per_process to improve oob performance

### PR DESCRIPTION
**Pull request by sywangyi**
Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

### CODE REVIEW

**Comment by sywangyi** [comment]
@sgugger @yao-matrix @liangan1 please help review it

File: src/accelerate/commands/launch.py [hunk 1]
```diff
@@ -307,6 +309,20 @@ def simple_launcher(args):
         mixed_precision = "fp16"
 
     current_env["MIXED_PRECISION"] = str(mixed_precision)
+    if args.num_cpu_threads_per_process == 1:
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:312]: We shouldn't change this value if the user has set it to 1. Maybe change the default of `args.num_cpu_threads_per_process` to `None` and then set it to the right default if it's unset.

File: src/accelerate/commands/launch.py [hunk 2]
```diff
@@ -307,6 +309,20 @@ def simple_launcher(args):
...
+            ["MPI_LOCALNRANKS", "OMPI_COMM_WORLD_LOCAL_SIZE", "MV2_COMM_WORLD_LOCAL_SIZE"], 1
+        )
+        args.num_cpu_threads_per_process = int(psutil.cpu_count(logical=False) / local_size)
+        if args.num_cpu_threads_per_process == 0:
+            args.num_cpu_threads_per_process = 1
+        if args.num_cpu_threads_per_process != 1:
+            logger.warn(
+                'num_cpu_threads_per_process is reset to %d instead of default 1 to improve oob performance.\
+        \nyou could tun num_cpu_threads_per_process by pass --num_cpu_threads_per_process xxx to "accelerate launch "\n'
+                % args.num_cpu_threads_per_process
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:323]: Adapt the warning to the behavior suggested above (no default was provided, we picked xxx). Also use f-strings please.

**Review by sgugger — COMMENTED** [review]
Thanks for working on this!

File: src/accelerate/commands/launch.py (hunk 2 above)
Comment (sywangyi) [reply src/accelerate/commands/launch.py]: apply you suggestion @sgugger

File: src/accelerate/commands/launch.py [hunk 3]
```diff
@@ -337,6 +339,17 @@ def simple_launcher(args):
...
     current_env["MIXED_PRECISION"] = str(mixed_precision)
+    if args.num_cpu_threads_per_process is None:
+        local_size = get_int_from_env(
+            ["MPI_LOCALNRANKS", "OMPI_COMM_WORLD_LOCAL_SIZE", "MV2_COMM_WORLD_LOCAL_SIZE"], 1
+        )
+        args.num_cpu_threads_per_process = int(psutil.cpu_count(logical=False) / local_size)
+        if args.num_cpu_threads_per_process == 0:
+            args.num_cpu_threads_per_process = 1
+        logger.warn(
+            f"num_cpu_threads_per_process unset, we pick as {args.num_cpu_threads_per_process} to improve oob performance"
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:350]:
```suggestion
            f"num_cpu_threads_per_process unset, we set it at {args.num_cpu_threads_per_process} to improve oob performance."
```

File: src/accelerate/commands/launch.py [hunk 4]
```diff
@@ -337,6 +339,17 @@ def simple_launcher(args):
...
 
     current_env["MIXED_PRECISION"] = str(mixed_precision)
+    if args.num_cpu_threads_per_process is None:
+        local_size = get_int_from_env(
+            ["MPI_LOCALNRANKS", "OMPI_COMM_WORLD_LOCAL_SIZE", "MV2_COMM_WORLD_LOCAL_SIZE"], 1
+        )
+        args.num_cpu_threads_per_process = int(psutil.cpu_count(logical=False) / local_size)
+        if args.num_cpu_threads_per_process == 0:
+            args.num_cpu_threads_per_process = 1
+        logger.warn(
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:349]:
```suggestion
        logger.info(
```
Maybe the info level is enough?

File: src/accelerate/commands/launch.py [hunk 5]
```diff
@@ -434,7 +447,9 @@ def multi_gpu_launcher(args):
             current_env["FSDP_BACKWARD_PREFETCH"] = str(args.fsdp_backward_prefetch_policy)
         if args.fsdp_state_dict_type is not None:
             current_env["FSDP_STATE_DICT_TYPE"] = str(args.fsdp_state_dict_type)
-
+    if args.num_cpu_threads_per_process is None:
+        args.num_cpu_threads_per_process = 1
+        logger.warn(f"num_cpu_threads_per_process unset, we pick as {args.num_cpu_threads_per_process}")
```
Comment (sgugger) [inline src/accelerate/commands/launch.py:452]:
```suggestion
        logger.info(f"num_cpu_threads_per_process unset, we set it at {args.num_cpu_threads_per_process}.")
```

**Review by sgugger — APPROVED** [review]
Thanks for iterating! Left a couple more nits.

File: src/accelerate/commands/launch.py (hunk 4 above)
Comment (sywangyi) [reply src/accelerate/commands/launch.py]: change to info

**Pull request merged**